[ERE-1430] Upgrade to Bootstrap 5

ppettitau opened the pull request.

ppettitau committed:
[ERE-1430] Upgrade to Bootstrap 5

ScriptSmith committed:
Replace sidebar link parent panel selector with card selector

ppettitau committed:
[ERE-1430] Fix test selectors for cards following B5 upgrade.

ppettitau committed:
[ERE-1430] Always upload aloe artifacts even if tests fail

ppettitau committed:
[ERE-1430] Give the dropdown menu higher precedence to prevent the menu disappearing under the footer

ppettitau committed:
[ERE-1430] Tar test results before uploading to get around errors from having special characters in generated file names

ppettitau committed:
[ERE-1430] Fix progress bar label

ppettitau committed:
[ERE-1430] Fix progress bar selector

ppettitau committed:
[ERE-1430] Add padding to sidebar to match pre-upgrade look & feel
Modify e2e click method to use more stable scrolling function

ppettitau committed:
[ERE-1430] Increase VM Resolution
Maximise browser window
Add ability to pause after scrolling to an element
Cleanup history step and modify to use pause to get around issue with elements not being in the viewport to interact with them.

ppettitau committed:
[ERE-1430] Maximise browser window
Scroll elements into center of the window to avoid them being hidden behind banners
Only scroll to elements if they can't be clicked to minimise sleep times
Add stability to other areas

ppettitau committed:
[ERE-1430] Upgrade to bootstrap-slider 11.0.2 with better Bootstrap support (supports Bootstrap 4)
Add hack to get the slider to work with Bootstrap 5
Other general fixes

ppettitau committed:
[ERE-1430] Fix styles for multi file uploads
Fix bug with conditional rendering fields

ppettitau committed:
[ERE-1430] Fix bug with printing on Chrome

ppettitau committed:
[ERE-1430] Fix styling for clear option when multiple files have been saved.

ppettitau committed:
[ERE-1430] Update print buttons to be consistently styled

ppettitau committed:
[ERE-1430]
Bootstrap upgrade for registration forms.
Update 'required' asterisk to no longer be displayed as an input's background as this clashes with Bootstrap 5 styling, particularly dropdowns (select).

ppettitau committed:
[ERE-1430] Fix changed attribute name for Bootstrap Modal

ppettitau committed:
[ERE-1430] Add the missing option for Indeterminate gender when adding a child/patient.

ScriptSmith committed:
Replace navbar hamburger with animated chevron

ScriptSmith committed:
Update bootstrap-slider version reference

ScriptSmith commented on rdrf/explorer/templates/explorer/query.html:
```diff
@@ -192,49 +192,45 @@
         <button id="hide-description-btn" type="button" class="btn btn-primary btn-sm">{% trans "Hide description" %}</button>
```
Have a look at `ReportDataTableView` / `report_table_view.html` as well. The buttons generated by datatables aren't being formatted correctly.

ppettitau replied:
Can you show me how to use this functionality?

ScriptSmith replied:
Sure

ScriptSmith replied:
Seems it was a caching issue as well, but I'll take you through the interface

ScriptSmith commented on rdrf/rdrf/static/css/bootstrap-overrides.css:
```diff
@@ -0,0 +1,72 @@
+/* Fix the admin menu that gets cut off to the right of the page since Bootstrap 5 no longer has .navbar-right styles. */
```
I think these should eventually be moved into the main stylesheet / page-appropriate styles, perhaps as new classes.

It also seems counterproductive to fight bootstrap's defaults too much, so we might also want to examine if there's a more 'idiomatic' way to present the elements we've had to modify.

ScriptSmith commented on rdrf/rdrf/templates/rdrf_cdes/patients_listing.html:
```diff
@@ -127,7 +127,7 @@
                     columns: {{columns|safe }}
```
The column heading icons seem to be using a glyphicon font that has since been removed, so they aren't being rendered correctly.

ppettitau replied:
It is rendering for me. Maybe try a hard refresh?
![image](https://user-images.githubusercontent.com/60163872/133340296-8fc952bf-c00f-4116-8dbb-b65091bd7429.png)

ScriptSmith replied:
Yep, my bad!

ScriptSmith commented on rdrf/rdrf/templates/rdrf_cdes/navbar.html:
```diff
@@ -21,7 +21,7 @@
       </a>
       <button type="button" class="navbar-toggler" data-bs-toggle="collapse" data-bs-target="#navbar-collapsing-links" aria-controls="navbar-collapsing-links" aria-expanded="false" aria-label="Toggle navigation">
-          <span class="fa fa-bars"></span>
+          <span class="fa fa-chevron-down"></span>
```
What do you think of replacing the navbar hamburger with a chevron like this?

ppettitau replied:
I think the chevron looks good & clean. My only concern is that we are deviating from the "universally recognised" icon for a menu button in the mobile context. But in saying that, a chevron isn't too obscure it would probably work just as well.

ppettitau replied:
Ned to update the navbar on the landing page for consistency.

ScriptSmith replied:
Yeah, maybe it's worth finding a more distinct icon for the form links instead?

ScriptSmith replied:
Also, one or both of the icons could be accompanied by text labels for clarity.

ppettitau committed:
[ERE-1430] Sidebar menu button
e2e - Removed exception handling workaround from click method and implement explicit scroll where required.

ppettitau committed:
[ERE-1430] Revert changes to e2e tests as they are failing.

ppettitau committed:
[ERE-1430] Remove xpath references to the sidebar, and locate by ID for improved clarity and performance.

ppettitau committed:
[ERE-1430] Fix mis-alignment of sidebar

ppettitau committed:
[ERE-1430] Fix padding of sidebar

ScriptSmith committed:
Attach changelist tooltips to cell

ScriptSmith committed:
Merge branch 'next_release' into ERE-1430_upgrade_to_Bootstrap5

ppettitau committed:
Fix logo styling
